include/swift/SILOptimizer/Utils/InstOptUtils.h

```diff
@@ -384,6 +384,16 @@ struct LLVM_LIBRARY_VISIBILITY FindLocalApplySitesResult {
 Optional<FindLocalApplySitesResult>
 findLocalApplySites(FunctionRefBaseInst *fri);
 
+/// Return true iff the given operand, that is a non-trivial and non-address
+/// value, is consumed by its user. That means, in ownership SIL this use must
+/// consume the value, and in non-ownership SIL, this operand must be released
+/// by this use. The given operand is required to be non-trivial and
+/// non-address.
+///
+/// TODO: Review the semantics of operations that extend the lifetime *without*
+/// propagating the value. Ideally, that never happens without borrowing first.
```

Comment:
I don't have a good answer for this, but I am worried about finding a way to lexically disambiguate APIs that are only available in ownership SSA and ones in non-ownership SSA. I guess that asserts will guide us.

(The reason I am bringing this up is I almost asked you to put this into OwnershipUtils.h until I read the comment).

Reply:
I agree. Asserts and perhaps we can come up with a ownership-neutral naming convention for such operations. I was split between this and something like `doesDecrementReferenceCount`  but `isConsuming` seemed much more succinct and nicer in this context, though it comes with an  OSSA connotation

Reply:
Yeah, I also would expect such an API to be in OwnershipUtils. Eventually most of our utilities will work on ownership, maybe exclusively. So regardless of which file this API belongs in, it's still good to have a clear doc-comment convention and a convention for asserting at the top of the implementation.